frontend/subrequests/lint/lint.go

```diff
@@ -125,46 +118,38 @@ func (results *LintResults) ToResult() (*client.Result, error) {
 }
 
 func PrintLintViolations(dt []byte, w io.Writer) error {
-	var warnings LintResults
+	var results LintResults
 
-	if err := json.Unmarshal(dt, &warnings); err != nil {
+	if err := json.Unmarshal(dt, &results); err != nil {
 		return err
 	}
 
-	// Here, we're grouping the warnings by rule name
-	lintWarnings := make(map[string][]Warning)
-	lintWarningRules := []string{}
-	for _, warning := range warnings.Warnings {
-		if _, ok := lintWarnings[warning.RuleName]; !ok {
-			lintWarningRules = append(lintWarningRules, warning.RuleName)
-			lintWarnings[warning.RuleName] = []Warning{}
+	sort.Slice(results.Warnings, func(i, j int) bool {
+		sourceInfoI := results.Sources[results.Warnings[i].Location.SourceIndex]
+		sourceInfoJ := results.Sources[results.Warnings[j].Location.SourceIndex]
+		if sourceInfoI.Filename != sourceInfoJ.Filename {
+			return sourceInfoI.Filename < sourceInfoJ.Filename
 		}
-		lintWarnings[warning.RuleName] = append(lintWarnings[warning.RuleName], warning)
-	}
-	sort.Strings(lintWarningRules)
-
-	tw := tabwriter.NewWriter(w, 0, 0, 2, ' ', 0)
-	for _, rule := range lintWarningRules {
-		fmt.Fprintf(tw, "Lint Rule %s\n", rule)
-		for _, warning := range lintWarnings[rule] {
-			source := warnings.Sources[warning.Location.SourceIndex]
-			sourceData := bytes.Split(source.Data, []byte("\n"))
-			firstRange := warning.Location.Ranges[0]
-			if firstRange.Start.Line != firstRange.End.Line {
-				fmt.Fprintf(tw, "\t%s:%d-%d\n", source.Filename, firstRange.Start.Line, firstRange.End.Line)
-			} else {
-				fmt.Fprintf(tw, "\t%s:%d\n", source.Filename, firstRange.Start.Line)
-			}
-			fmt.Fprintf(tw, "\t%s\n", warning.Detail)
-			for _, r := range warning.Location.Ranges {
-				for i := r.Start.Line; i <= r.End.Line; i++ {
-					fmt.Fprintf(tw, "\t%d\t|\t%s\n", i, sourceData[i-1])
-				}
-			}
-			fmt.Fprintln(tw)
+		if len(results.Warnings[i].Location.Ranges) == 0 || len(results.Warnings[j].Location.Ranges) == 0 {
+			return false
 		}
-		fmt.Fprintln(tw)
-	}
+		return results.Warnings[i].Location.Ranges[0].Start.Line < results.Warnings[j].Location.Ranges[0].Start.Line
```

Comment:
just in case, I think we should protect against "index out of bounds" for `Ranges` in here.